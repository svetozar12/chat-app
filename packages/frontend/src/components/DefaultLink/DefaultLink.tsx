import { Center, Link as AnchorLink } from "@chakra-ui/react";
import React from "react";
import Link from "next/dist/client/link";

interface IDefaultLink {
  text: string;
  href: string;
}

const DefaultLink = ({ text, href }: IDefaultLink) => {
  return (
    <Center w="full">
      <Link href={href}>
        <AnchorLink textAlign="center" w="full" color="blue.500" fontWeight="bold" href="#">
          {text}
        </AnchorLink>
      </Link>
    </Center>
  );
};

export default DefaultLink;
