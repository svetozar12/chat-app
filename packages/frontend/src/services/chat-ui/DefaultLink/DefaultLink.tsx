// eslint-disable-next-line import/named
import { Center, CenterProps, Link as AnchorLink } from '@chakra-ui/react';
import React from 'react';
import Link from 'next/dist/client/link';
import { IBaseComponent } from 'services/chat-ui/types';

type Base = IBaseComponent<CenterProps>;
interface IDefaultLink extends Base {
  text: string;
  href: string;
}

function DefaultLink(props: IDefaultLink) {
  const { href, text, baseProps, chakraProps, style } = props;
  return (
    <Center w="full" {...chakraProps} {...style} {...baseProps}>
      <Link href={href}>
        <AnchorLink textAlign="center" w="full" color="blue.500" fontWeight="bold" href="#">
          {text}
        </AnchorLink>
      </Link>
    </Center>
  );
}

export default DefaultLink;
