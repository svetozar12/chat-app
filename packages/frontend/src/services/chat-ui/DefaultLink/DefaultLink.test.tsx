import { mount, ReactWrapper } from 'enzyme';
import DefaultLink from 'services/chat-ui/DefaultLink/DefaultLink';
import Link from 'next/link';
import { Link as ArchorLink } from '@chakra-ui/react';

const componentProps = { text: 'HomePage', href: 'http://go_to_home_page' };
const baseProps = { textAlign: 'center', w: 'full', color: 'blue.500', fontWeight: 'bold', href: '#' };
const setupMount = () => {
  return mount(<DefaultLink {...componentProps} />);
};

describe(`testing correct behaviour of <DefaultLink />`, () => {
  let wrapper: ReactWrapper;
  beforeAll(() => {
    wrapper = setupMount();
  });
  it('component <DefaultLink /> should render', () => {
    expect(wrapper.find(DefaultLink).length).toBe(1);
  });
  it(`component <DefaultLink /> should have href with value ${componentProps.href}`, () => {
    expect(wrapper.find(Link).prop('href')).toBe(componentProps.href);
  });
  it(`component <DefaultLink /> should have text with value ${componentProps.text}`, () => {
    expect(wrapper.find(ArchorLink).text()).toBe(componentProps.text);
  });
  Object.entries(baseProps).forEach(([key, value]) => {
    it(`component <DefaultLink /> should have prop ${key}:${value}`, () => {
      expect(wrapper.find(ArchorLink).prop(key)).toBe(value);
    });
  });
  it('component <DefaultLink /> should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
