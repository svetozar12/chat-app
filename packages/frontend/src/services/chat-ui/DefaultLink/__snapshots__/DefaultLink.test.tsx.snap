// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing correct behaviour of <DefaultLink /> component <DefaultLink /> should match snapshot 1`] = `
<DefaultLink
  href="http://go_to_home_page"
  text="HomePage"
>
  <Center
    w="full"
  >
    <Noop />
    <div
      className="css-0"
    >
      <ForwardRef(LinkComponent)
        href="http://go_to_home_page"
      >
        <Link
          color="blue.500"
          fontWeight="bold"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          textAlign="center"
          w="full"
        >
          <Styled(a)
            __css={Object {}}
            className="chakra-link"
            color="blue.500"
            fontWeight="bold"
            href="#"
            onClick={[Function]}
            onMouseEnter={[Function]}
            textAlign="center"
            w="full"
          >
            <Noop />
            <a
              className="chakra-link css-0"
              href="#"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              HomePage
            </a>
          </Styled(a)>
        </Link>
      </ForwardRef(LinkComponent)>
    </div>
  </Center>
</DefaultLink>
`;
